internal/gcsx/read_manager/read_manager.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package read_manager
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/common"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/cache/file"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/gcsx"
+	clientReaders "github.com/googlecloudplatform/gcsfuse/v2/internal/gcsx/client_readers"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
+)
+
+type ReadManager struct {
+	gcsx.ReadManager
+	object *gcs.MinObject
+
+	// readers holds a list of data readers, prioritized for reading.
+	// e.g., File cache reader, GCS reader.
+	readers []gcsx.Reader
+}
+
+// NewReadManager creates a new ReadManager for the given GCS object.
+// It initializes the manager with a file cache reader and a GCS reader.
+func NewReadManager(o *gcs.MinObject, bucket gcs.Bucket, sequentialReadSizeMB int32, fileCacheHandler *file.CacheHandler, cacheFileForRangeRead bool, metricHandle common.MetricHandle, mrdWrapper *gcsx.MultiRangeDownloaderWrapper) *ReadManager {
+	gcsReader := clientReaders.NewGCSReader(o, bucket, metricHandle, mrdWrapper, sequentialReadSizeMB)
+	fileCacheReader := gcsx.NewFileCacheReader(o, bucket, fileCacheHandler, cacheFileForRangeRead, metricHandle)
```

Comment:
I see, fileCacheReader is created every time. Or we want to create only when file-cache is enabled.?

Reply:
I don't think that's an expensive operation. I'm creating a new range reader in the test specifically for scenarios where I want to disable the file cache. Doing the opposite seems like it would involve adding more code to each test.

Reply:
Added conditional check that if file cache handler is passed then only add file cache reader.